### ISSUE TITLE
Version mismatch check failing during TS compilation

### ISSUE THREAD

**Issue by Sikarii** [edgedb/edgedb-js#434]
**Code**
[src/reflection/generate.ts](https://github.com/edgedb/edgedb-js/blob/519726df6aa5025a7d35bb23a61cd313478d6486/src/reflection/generate.ts#L149-L162)

```typescript
const { $, _edgedbJsVersion } = require("edgedb");

if (_edgedbJsVersion !== "0.21.2") {
  throw new Error(
    `The query builder was generated by a different version of edgedb-js (v0.21.2)` +
      ` than the one currently installed (v${_edgedbJsVersion}).\n` +
      `Run 'npx edgeql-js' to re-generate a compatible version.\n`
  );
}
```

**Error or desired behavior**
The above in the query builder code fails due to TS2367:
`TS2367: This condition will always return 'true' since the types '"0.21.3"' and '"0.21.2"' have no overlap`.

This will cause the error to be very confusing to users, instead of throwing the useful and intended runtime error.

---

**I would be willing to submit a PR for this**.
Would casting `_edgedbJsVersion` to `string` to "de-constify" fix this or is there a better way?

Arguably, the compile time error is better than having it throw runtime, is it possible to improve the error during compile time?

**Comment by colinhacks** [comment]
This is already fixed in v0.22. I deconstified it because it's not really clear how'd we'd improve the compile time error. https://github.com/edgedb/edgedb-js/commit/d5ea4141da6ad3dd54f47526954b1c4fe3b4df0a#diff-77e96ffbfe37e038b568e238592bd039971183b0899cd16f70c7b27e0bcff569R47

**Comment by Sikarii** [comment]
Ah my bad, I clearly need up my git blame skills!
